hadoop-hdds/server-scm/src/main/java/org/apache/hadoop/hdds/scm/pipeline/PipelineManagerV2Impl.java

```diff
@@ -420,10 +392,50 @@ public void scrubPipeline(ReplicationType type, ReplicationFactor factor)
           " since it stays at ALLOCATED stage for " +
           Duration.between(currentTime, p.getCreationTimestamp()).toMinutes() +
           " mins.");
-      finalizeAndDestroyPipeline(p, false);
+      closePipeline(p, false);
+      closeContainersForPipeline(p.getId());
     }
   }
 
+  private void scrubClosedPipeline(
+      ReplicationType type, ReplicationFactor factor, Instant currentTime)
+      throws IOException {
+    long pipelineDestroyTimeoutInMillis =
+        conf.getTimeDuration(ScmConfigKeys.OZONE_SCM_PIPELINE_DESTROY_TIMEOUT,
+            ScmConfigKeys.OZONE_SCM_PIPELINE_DESTROY_TIMEOUT_DEFAULT,
+            TimeUnit.MILLISECONDS);
+    List<Pipeline> closedPipelines = stateManager.getPipelines(type, factor,
+        Pipeline.PipelineState.CLOSED).stream()
```

Comment:
Should we avoid the stream api for better performance ?

Reply:
I updated with for loop